### PR TITLE
Another batch of v6 regressions

### PR DESCRIPTION
**Pull request by MaceWindu**

### CODE REVIEW

File: Source/LinqToDB/Linq/Builder/AggregationBuilder.cs [hunk 1]
```diff
@@ -318,7 +318,7 @@ internal class AggregationInfo
 				valueExpression = SequenceHelper.PrepareBody(inputValueLambda, currentRef.BuildContext);
 			}
 
-			if (aggregationType != AggregationType.Count || isDistinct)
+			if (aggregationType != AggregationType.Custom && aggregationType != AggregationType.Count || isDistinct)
```
Comment (viceroypenguin) [inline Source/LinqToDB/Linq/Builder/AggregationBuilder.cs:321]:
```suggestion
			if (aggregationType is not (AggregationType.Custom or AggregationType.Count) || isDistinct)
```

File: Source/LinqToDB/Linq/Builder/ProjectFlagExtensions.cs [hunk 2]
```diff
@@ -150,5 +150,13 @@ public static bool IsForExtension(this ProjectFlags flags)
 			return (flags & ProjectFlags.ForExtension) != 0;
 		}
 
+		[DebuggerStepThrough]
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static bool IsForceOuter(this ProjectFlags flags)
+		{
+			return (flags & ProjectFlags. ForceOuterAssociation) != 0;
```
Comment (viceroypenguin) [inline Source/LinqToDB/Linq/Builder/ProjectFlagExtensions.cs:157]:
```suggestion
			return (flags & ProjectFlags.ForceOuterAssociation) != 0;
```

**Review by viceroypenguin — APPROVED** [review]

**Comment by MaceWindu** [comment]
Looks good except couple of suspicious baselines changes:
- `DefaultIfEmptyTests.WithDefaultInSelectMany`
- `EagerLoadingTests.TestCardinalityProjectedOptional1` + `EagerLoadingTests.TestCardinalityProjectedOptional3`
- probably `CteTests.TestCustomCount`
- aliases changed for some tests

**Pull request merged**